src/py/flwr/server/superlink/fleet/grpc_rere/server_interceptor.py

```diff
@@ -79,22 +80,22 @@ def _get_value_from_tuples(
 class AuthenticateServerInterceptor(grpc.ServerInterceptor):  # type: ignore
     """Server interceptor for client authentication."""
 
-    def __init__(
-        self,
-        client_public_keys: Set[bytes],
-        private_key: ec.EllipticCurvePrivateKey,
-        public_key: ec.EllipticCurvePublicKey,
-    ):
-        self.server_private_key = private_key
-        self.client_public_keys = client_public_keys
-        self.encoded_server_public_key = base64.urlsafe_b64encode(
-            public_key_to_bytes(public_key)
-        )
-        log(
-            INFO,
-            "Client authentication enabled with %d known public keys",
-            len(client_public_keys),
-        )
+    def __init__(self, state: State):
+        self.state = state
+        self._lock = threading.Lock()
+
+        self.client_public_keys = state.get_client_public_keys()
+        if len(self.client_public_keys) == 0:
+            raise ValueError("Known client public keys must be at least 1")
```

Comment:
Let's make it a warning message instead of an error. We'll have to support `0` once we make this dynamic.

Reply:
Reverted.